src/Univapay/Resources/Subscription/ScheduledPayment.php

```diff
@@ -49,6 +49,7 @@ public function __construct(
         $this->amountFormatted = $amountFormatted;
         $this->isPaid = $isPaid;
         $this->isLastPayment = $isLastPayment;
+        $createdOn = $createdOn ?? date('now');
```

Comment:
Hmm, is this necessary? Because createdOn is always provided by the API response and is not intended for users to manually create instances of the object for themselves...

Reply:
depracated warning on php 8 and above
<img width="683" alt="Screenshot 2024-11-12 at 15 38 01" src="https://github.com/user-attachments/assets/5a670bb6-884b-4f5e-9ad7-cc671e5f1c52">

Reply:
ah i see, i saw some deprecated warning during the test as we pass null to `date_create()`.
But looking here, i need to check it again on why we get null in the 1st place

Reply:
We use this class for 2 different test (ScheduledPaymentTest and SubscriptionTest), the warning came from SubscriptionTest (endpoint: `/subscriptions/simulate_plan`) which return below response
```
[
    {
        "due_date": "xxx-xx-xx",
        "zone_id": "xxxx/xxxx",
        "amount": xxxxxx,
        "currency": "xxx",
        "is_paid": xxxx,
        "is_last_payment": xxxx
    }
]
```

I think it should be fine to use this setup, as in the end it we just adding default value for a case when the field should not exists in the 1st place.
Note: by default the `create_date` will use [now](https://www.php.net/manual/en/function.date-create.php) as default if we pass a null

Warning on PHP 8+
<img width="931" alt="Screenshot 2024-11-12 at 18 01 27" src="https://github.com/user-attachments/assets/179377f4-dc6a-4182-b9ee-e1c50cd7e72b">

Reply:
Ahh... Okies.. add a comment in there so we know that it is because of simulate plan.

Reply:
the previous one was a mistake, the string one is the correct one.

Reply:
I didn't know you can write statements in the middle of a constructor? 😕

Reply:
it was handled inside the date_create behind,
i was testing it on php layer tested it sending with `null` and with `now` and it gave me same value
if we took at the code behind it 
```
/* {{{ Returns new DateTime object */
PHP_FUNCTION(date_create)
{
	zval           *timezone_object = NULL;
	char           *time_str = NULL;
	size_t          time_str_len = 0;

	ZEND_PARSE_PARAMETERS_START(0, 2)
		Z_PARAM_OPTIONAL
		Z_PARAM_STRING(time_str, time_str_len)
		Z_PARAM_OBJECT_OF_CLASS_OR_NULL(timezone_object, date_ce_timezone)
	ZEND_PARSE_PARAMETERS_END();

	php_date_instantiate(date_ce_date, return_value);
	if (!php_date_initialize(Z_PHPDATE_P(return_value), time_str, time_str_len, NULL, timezone_object, 0)) {
		zval_ptr_dtor(return_value);
		RETURN_FALSE;
	}
}
/* }}} */
```
it still do nothing until the. php_date_initialize which is below
```
PHPAPI int php_date_initialize(php_date_obj *dateobj, const char *time_str, size_t time_str_len, const char *format, zval *timezone_object, int flags) /* {{{ */
{
	timelib_time   *now;
	timelib_tzinfo *tzi = NULL;
	timelib_error_container *err = NULL;
	int type = TIMELIB_ZONETYPE_ID, new_dst = 0;
	char *new_abbr = NULL;
	timelib_sll new_offset = 0;
	time_t sec;
	suseconds_t usec;
	int options = 0;

	if (dateobj->time) {
		timelib_time_dtor(dateobj->time);
	}
	if (format) {
		if (time_str_len == 0) {
			time_str = "";
		}
		dateobj->time = timelib_parse_from_format(format, time_str, time_str_len, &err, DATE_TIMEZONEDB, php_date_parse_tzfile_wrapper);
	} else {
		if (time_str_len == 0) {
			time_str = "now";
			time_str_len = sizeof("now") - 1;
		}
		dateobj->time = timelib_strtotime(time_str, time_str_len, &err, DATE_TIMEZONEDB, php_date_parse_tzfile_wrapper);
	}

	/* update last errors and warnings */
	update_errors_warnings(err);


	if ((flags & PHP_DATE_INIT_CTOR) && err && err->error_count) {
		/* spit out the first library error message, at least */
		php_error_docref(NULL, E_WARNING, "Failed to parse time string (%s) at position %d (%c): %s", time_str,
			err->error_messages[0].position, err->error_messages[0].character, err->error_messages[0].message);
	}
	if (err && err->error_count) {
		timelib_time_dtor(dateobj->time);
		dateobj->time = 0;
		return 0;
	}

	if (timezone_object) {
		php_timezone_obj *tzobj;

		tzobj = Z_PHPTIMEZONE_P(timezone_object);
		switch (tzobj->type) {
			case TIMELIB_ZONETYPE_ID:
				tzi = tzobj->tzi.tz;
				break;
			case TIMELIB_ZONETYPE_OFFSET:
				new_offset = tzobj->tzi.utc_offset;
				break;
			case TIMELIB_ZONETYPE_ABBR:
				new_offset = tzobj->tzi.z.utc_offset;
				new_dst    = tzobj->tzi.z.dst;
				new_abbr   = timelib_strdup(tzobj->tzi.z.abbr);
				break;
		}
		type = tzobj->type;
	} else if (dateobj->time->tz_info) {
		tzi = dateobj->time->tz_info;
	} else {
		tzi = get_timezone_info();
	}

	now = timelib_time_ctor();
	now->zone_type = type;
	switch (type) {
		case TIMELIB_ZONETYPE_ID:
			now->tz_info = tzi;
			break;
		case TIMELIB_ZONETYPE_OFFSET:
			now->z = new_offset;
			break;
		case TIMELIB_ZONETYPE_ABBR:
			now->z = new_offset;
			now->dst = new_dst;
			now->tz_abbr = new_abbr;
			break;
	}
	php_date_get_current_time_with_fraction(&sec, &usec);
	timelib_unixtime2local(now, (timelib_sll) sec);
	php_date_set_time_fraction(now, usec);

	options = TIMELIB_NO_CLONE;
	if (flags & PHP_DATE_INIT_FORMAT) {
		options |= TIMELIB_OVERRIDE_TIME;
	}
	timelib_fill_holes(dateobj->time, now, options);

	timelib_update_ts(dateobj->time, tzi);
	timelib_update_from_sse(dateobj->time);

	dateobj->time->have_relative = 0;

	timelib_time_dtor(now);

	return 1;
} /* }}} */
```

Reply:
Ahh, never mind.. its in the constructor block.. I thought it was in the constructor params... github hid the part such that it looked like it was in the constrictor params.